pkg/quota/admission/runonceduration/admission.go

```diff
@@ -114,14 +114,27 @@ func (a *runOnceDuration) applyProjectAnnotationOverride(namespace string, pod *
 	if ns.Annotations == nil {
 		return false, nil
 	}
-	override, hasOverride := ns.Annotations[api.ActiveDeadlineSecondsOverrideAnnotation]
-	if !hasOverride {
+	limit, hasLimit := ns.Annotations[api.ActiveDeadlineSecondsLimitAnnotation]
+	if !hasLimit {
 		return false, nil
 	}
-	overrideInt64, err := strconv.ParseInt(override, 10, 64)
+	limitInt64, err := strconv.ParseInt(limit, 10, 64)
 	if err != nil {
-		return false, fmt.Errorf("cannot parse the ActiveDeadlineSeconds override (%s) for project %s: %v", override, ns.Name, err)
+		return false, fmt.Errorf("cannot parse the ActiveDeadlineSeconds limit (%s) for project %s: %v", limit, ns.Name, err)
 	}
-	pod.Spec.ActiveDeadlineSeconds = &overrideInt64
+	pod.Spec.ActiveDeadlineSeconds = minInt64(&limitInt64, pod.Spec.ActiveDeadlineSeconds)
 	return true, nil
 }
+
+func minInt64(a, b *int64) *int64 {
```

Comment:
fyi, there is https://github.com/kubernetes/kubernetes/blob/master/pkg/util/integer/integer.go